crates/suiop-cli/src/cli/ci/image.rs

```diff
@@ -0,0 +1,263 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::cli::lib::{get_api_server, get_oauth_token};
+use anyhow::Result;
+
+use chrono::{DateTime, Local, Utc};
+use clap::{Parser, ValueEnum};
+use colored::Colorize;
+use serde::{self, Serialize};
+use std::{fmt::Display, str::FromStr};
+use tabled::{settings::Style, Table, Tabled};
+use tracing::debug;
+
+#[derive(Tabled)]
+struct BuildInfo {
+    name: String,
+    status: String,
+    #[tabled(rename = "Start Time (Local Time)")]
+    start_time: String,
+    #[tabled(rename = "End Time")]
+    end_time: String,
+}
+
+#[derive(Parser, Debug)]
+pub struct ImageArgs {
+    #[command(subcommand)]
+    action: ImageAction,
+}
+
+#[derive(ValueEnum, Clone, Debug)]
+#[clap(rename_all = "lowercase")]
+pub enum RefType {
+    Branch,
+    Tag,
+    Commit,
+}
+
+impl Serialize for RefType {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(match self {
+            RefType::Branch => "branch",
+            RefType::Tag => "tag",
+            RefType::Commit => "commit",
+        })
+    }
+}
+
+impl Display for RefType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            RefType::Branch => write!(f, "branch"),
+            RefType::Tag => write!(f, "tag"),
+            RefType::Commit => write!(f, "commit"),
+        }
+    }
+}
+
+#[derive(clap::Subcommand, Debug)]
+pub enum ImageAction {
+    #[command(name = "build")]
+    Build {
+        #[arg(short, long)]
+        repo_name: String,
+        #[arg(short, long)]
+        dockerfile: String,
+        #[arg(long)]
+        image_tag: Option<String>,
+        #[arg(long)]
```

Comment:
Can we use [`default_value` here](https://docs.rs/clap/latest/clap/struct.Arg.html#method.default_value) defined with a fn to somehow use the repo name? I don't know if that's possible. 
```suggestion
        /// the name of the image to use (defaults to repo name)
        #[arg(long)]
```

Reply:
Here's what chatgpt had to say, looks pretty involved but not impossible 

In Rust, using the `clap` crate, you can define default values for arguments. However, setting a default value for one argument based on another argument's value is a bit more involved. You would need to manually handle this logic after parsing the command line arguments.

Here's how you can achieve this:

1. Parse the command line arguments using `clap`.
2. Check if the argument with the dependent default value is provided.
3. If not, set it to the value of the other argument.

Let's go through an example:

### Step-by-Step Example

First, add the necessary dependencies to your `Cargo.toml`:

```toml
[dependencies]
clap = { version = "4.0", features = ["derive"] }
```

### Define and Parse Arguments

Here's an example of how to define and parse the arguments:

```rust
use clap::{Arg, Command};

fn main() {
    let matches = Command::new("example")
        .version("1.0")
        .author("Your Name <your.email@example.com>")
        .about("An example of default arguments in clap")
        .arg(
            Arg::new("input")
                .long("input")
                .takes_value(true)
                .help("Sets the input file")
                .required(true),
        )
        .arg(
            Arg::new("output")
                .long("output")
                .takes_value(true)
                .help("Sets the output file"),
        )
        .get_matches();

    let input = matches.value_of("input").unwrap();
    let output = matches.value_of("output").unwrap_or(input);

    println!("Input: {}", input);
    println!("Output: {}", output);
}
```

### Explanation

1. **Dependencies**: We use `clap` with the `derive` feature to simplify argument parsing.
2. **Command Definition**: We define a command with two arguments, `input` and `output`.
3. **Parsing**: We parse the arguments using `get_matches`.
4. **Default Logic**: We check if `output` is provided. If not, we set it to the value of `input`.

### Test Plan

To ensure the functionality works as expected, follow this test plan:

1. **Compilation Test**: Ensure the code compiles without errors by running `cargo build`.
2. **Execution Test**:
   - Run the command with both `--input` and `--output` arguments: `cargo run -- --input file1 --output file2`. Verify the output.
   - Run the command with only the `--input` argument: `cargo run -- --input file1`. Verify that the output is the same as the input.
3. **Error Handling**: Run the command without the required `--input` argument to ensure it fails with a meaningful error message.
4. **Edge Cases**:
   - Test with empty strings for `input` and `output`.
   - Test with very long file names.

### Clarifications Needed

To tailor the solution better, could you provide more details on:
1. **Argument Types**: Are the `input` and `output` arguments always strings (e.g., file paths), or do they have different types?
2. **Validation**: Do you need any specific validation on these arguments (e.g., checking if a file exists)?
3. **Default Behavior**: Should the default value logic be applied in all cases, or only under certain conditions?

These details will help in providing a more robust and fitting solution.